src/gcp.rs

```diff
@@ -199,24 +199,29 @@ impl GcpAuthenticator {
         match self {
             Self::Credentials(inner) => {
                 let expires_in = inner.token.read().unwrap().expires_in() as u64;
-                let mut deadline =
-                    Duration::from_secs(expires_in - METADATA_TOKEN_EXPIRY_MARGIN_SECS);
+                let mut deadline = Duration::from_secs(
+                    expires_in
+                        .saturating_sub(METADATA_TOKEN_EXPIRY_MARGIN_SECS)
+                        .max(1),
+                );
                 loop {
+                    debug!(
+                        "Sleeping for {}s before refreshing GCP authentication token",
+                        deadline.as_secs()
+                    );
```

Comment:
Debug can provide structured data:
```suggestion
                    debug!(%deadline, "Sleeping before refreshing GCP authentication token");
```
The wording now sounds awkward, but I'm not sure how to word it better.